src/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+project(CMaizeSuperbuild_core LANGUAGES CXX)
+
+add_library(${PROJECT_NAME} SHARED hello-world.cpp)
+
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER
+        ${CMAKE_CURRENT_SOURCE_DIR}/Shared-Library/hello-world.hpp
+        )
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
+        $<INSTALL_INTERFACE:include/>
+        )
+
+install(TARGETS ${PROJECT_NAME}
+        EXPORT ${PROJECT_NAME}-targets
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}
```

Comment:
Pretty sure you need `include(GNUInstallDirs)` before you can use `${CMAKE_INSTALL_INCLUDEDIR}`